enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,454 @@
+manifest is added for a component, an appropriate subset of existing
+tests for that component should ideally be run using the new
+manifest. If new configuration control items or features are added to
+a component, those items will require new tests.
+
+### Graduation Criteria
+
+This enhancement does not add or change any APIs or other user-visible
+features, so maturity levels and deprecation policies are not
+relevant.
+
+Graduation from phase 1 will happen when the SNC tooling is updated to
+use the new cluster profile. This should not result in any visible
+changes to users of the CRC product or to other users of SNC.
+
+As phase 2 is an open ended process of optimizing individual
+components, these changes will become available to SNC users,
+including CRC, as they are merged.
+
+### Upgrade / Downgrade Strategy
+
+The presence of the single-node-developer cluster profile, annotations
+for the profile on existing manifest files, and new specialized
+manifests annotated just for this profile will have no effect on
+clusters using any other cluster profile, including the default
+profile. Therefore, there should be no upgrade or downgrade concerns
+for any clusters not created via the SNC tooling.
+
+Clusters that are created via the SNC tooling are generally
+short-lived, and not upgradable across major OpenShift releases. No
+support is planned for upgrading or downgrading clusters to or from
+using the new profile.
+
+Upgrade / downgrade strategy may need to be addressed for new
+configuration items added to individual components to allow
+optimization for single node developer cluster usage.
+
+### Version Skew Strategy
+
+No component changes are foreseen as part of this enhancement that
+would require handling version skew.
+
+## Implementation History
+
+TBD
+
+## Drawbacks
+
+The main drawback of the proposed enhancement is that there may turn
+out to be multiple slightly different ideas of what constitutes a
+single node developer cluster. This could result in the proposed
+single-node-developer profile becoming a compromise that is not ideal
+for CRC or for any other of its use cases. But discouraging reuse
+(i.e. naming this the code-ready-containers profile) might result in a
+proliferation of similar but distinct cluster profiles each requiring
+independent maintenance and testing overhead. We hope to find a
+balance between these extremes, where the single-node-developer
+profile content is driven by the requirements of the CRC product, but
+is useful as-is in other cases as well. If this doesn’t turn out to be
+possible, additional profiles might utilize the many of the same
+specialized manifest files as the single-node-developer profile,
+minimizing the amount of additional maintenance and testing overhead.
+
+Another drawback is that any new cluster profile imposes a new
+supported topology on all components in the OpenShift payload,
+including all future platform extensions. Component teams will need to
```

Comment:
extensions are outside the payload, can you clarify what you mean by a sample extension you intend to highlight in CRC?

does CRC intent to limit the set of operators that appear in OLM for their use case that has been vetted to work?

Reply:
This text came from a review of a draft of the enhancement prior to submitting the PR. I don't think that review comment was about CRC-specific extensions, but instead was regarding extensions to OpenShift itself. Would it be clearer if I just drop "including all future platform extensions"? It seems to me that the same concerns apply to future components that are part of the payload and those that are external extensions to it.

I am not aware of any intent to limit what operators are are available via OLM in CRC clusters. Is the concern here that an operator may have a dependency on a component excluded in the profile? If so, could the required component be pulled in via the dependency, even if its manifest in the payload does not have the profile's annotation? Or should the operator not be available because its dependency was excluded?

Reply:
Added text about OLM and operator hub under Open Questions.